### ISSUE TITLE
404 Not Found on all api calls

### ISSUE THREAD

**Issue by ihunzai-dev** [Nexmo/nexmo-java#143]
I am getting 404 response on all API calls for the current in-progress voice call on local machine however it works fine on azure instance located in US region.

The number I purchased is US-Based, and my local machine is in Asia.  The issue is much like the one mentioned in this [post](https://help.nexmo.com/hc/en-us/articles/115015969628-Why-do-I-get-a-404-when-trying-to-change-an-active-conversation-)

There should be some option to set API base-url in client library

**Comment by cr0wst** [comment]
It looks like this functionality is going to be provided with #77 as it also resolves #58.  

@judy2k offered up a workaround on https://stackoverflow.com/questions/48685775/404-error-in-all-nexmo-api-calls in regards to extending the `ModifyCallMethod` class's `makeRequest` method to provide a new base url.

**Comment by cr0wst** [comment]
Resolved with #194 released in 3.9.0

**Comment by vinokanth92** [comment]
Hey I am trying to resolve this issue by configuring the baseUrl. According this [article ](https://help.nexmo.com/hc/en-us/articles/115015969628-Why-do-I-get-a-404-when-trying-to-change-an-active-conversation-) the 2 possible API base ULIs are 
1. https://api-us-1.nexmo.com
2. https://api-sg-1.nexmo.com

I configured them in the following way:
```
            HttpConfig httpConfig = new HttpConfig.Builder()
                    .apiBaseUri("https://api-sg-1.nexmo.com")
                    .build();
```
But I still do get 404 NOT FOUND for both the listed base URIs. 

When I call Nexmo API from my local server (Sydney) everything works. But when deploy it on AWS (USA) region I do get 404. How do I resolve this?

**Comment by cr0wst** [comment]
Hi @vinokanth92,

I think you found a bug that I didn't account for when implementing the base uri rewriting logic. I'll try to get a patch out today to fix this.

The full URI that gets used for some of our API endpoints is `https://api.nexmo.com/v1/...` and I was stripping out the `v1` if you weren't electing to use the default uri. Sorry about that!

Try and see if using `https://api-sg-1.nexmo.com/v1` fixes it in the meantime.

Steve

**Comment by vinokanth92** [comment]
Hey Steve thanks for your response. There is no need to apologise. You guys
are doing a great job. As you suggested adding v1 works like a charm.
Thanks a lot. Cheers.

**Comment by cr0wst** [comment]
Resolved in #217 released in `3.10.0`. The path for versioned APIs will always be `/v#/...` now instead of trying to guess when it would be appropriate.

I started the release about 20 minutes ago, it can take a bit before it hits the central repositories.

**Comment by vinokanth92** [comment]
@cr0wst Hello. I am currently using library version 4.01. Previously was using version 3.9.x and your solution of appending "v1" was working. But now when upgraded my library version this fix no more works. I get 404 for using both
- https://api-us-1.nexmo.com/v1
- https://api-sg-1.nexmo.com/v2

How can I fix this?
